### PR TITLE
WIP: ERSSUP-54768 - OAS Get Referral Request

### PR DESCRIPTION
**Pull request by nhsd-mark-warren**
See 
- https://nhsd-jira.digital.nhs.uk/browse/ERSSUP-54768 (task)
- https://nhsd-jira.digital.nhs.uk/browse/ERSSUP-54848 (overview)
- https://nhsd-jira.digital.nhs.uk/browse/ERSSUP-54880 (examples)

### CODE REVIEW

File: sandbox/src/mocks/getReferralRequest/responses/Unbooked.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  "id": "000000070000",
+  "meta": {
+    "versionId": "3",
+    "profile": [
+      "http://fhir.nhs.uk/StructureDefinition/ers-referralrequest-1"
+    ]
+  },
+  "resourceType": "ReferralRequest",
+  "contained": [
+    {
```
Comment (nhsd-mark-warren) [inline sandbox/src/mocks/getReferralRequest/responses/Unbooked.json:11]: is there any convention on the order in which information elements should be presented in the examples?

File: sandbox/src/mocks/getReferralRequest/responses/Unbooked.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                "valueReference": {
+                  "reference": "#Practitioner-021600556514"
+                }
+              }
+            ],
+            "contentType": "text/plain",
+            "url": "Binary/att-70000-70002",
+            "size": 4,
+            "title": "referral_letter.txt",
+            "creation": "ignoreDate"
```
Comment (nhsd-hubert-raich) [inline sandbox/src/mocks/getReferralRequest/responses/Unbooked.json:61]: We can't take expected json examples directly from regression pack, due to some fields being ignored there.

File: sandbox/src/mocks/getReferralRequest/responses/Unbooked.json [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      "resourceType": "Patient",
+      "identifier": [
+        {
+          "system": "http://fhir.nhs.uk/Id/nhs-number",
+          "value": "9912003888"
+        }
+      ]
+    },
+    {
+      "id": "DocumentReference-70002",
```
Comment (nhsd-mark-warren) [inline sandbox/src/mocks/getReferralRequest/responses/Unbooked.json:27]: Is there a conventional Document Reference number to use, just like NHS number? (I inherited this value from the "unbooked.json" XAPI test I based this example on)

File: sandbox/src/mocks/getReferralRequest/responses/Unbooked.json [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      "meta": {
+        "profile": [
+          "http://fhir.nhs.uk/StructureDefinition/ers-documentreference-1"
```
Comment (nhsd-mark-warren) [inline sandbox/src/mocks/getReferralRequest/responses/Unbooked.json:30]: I guess this should be replaced with a `$ref` to a schema in fhir/structure?

File: specification/e-referrals-service-api.yaml [hunk 5]
```diff
@@ -299,10 +299,68 @@ paths:
...
+        ## Overview
+
+        Use this endpoint to retrieve details of a referral. This includes references to clinical attachments, related referrals and other important data.
+
+        ## Pre-requisites
+        In order to use this endpoint you must be an authenticated e-RS user and use one of the following e-RS roles:
+          - `REFERRING_CLINICIAN`
+          - `REFERRING_CLINICIAN_ADMIN`
+          - `SERVICE_PROVIDER_CLINICIAN`
+          - `SERVICE_PROVIDER_CLINICIAN_ADMIN`
```
Comment (nhsd-hubert-raich) [inline specification/e-referrals-service-api.yaml:318]: SERVICE_PROVIDER_CLINICIAN_ADMIN will have to be removed as we don't support it right now (even if the overview spec still has this in ) . I've raised this with Ba's already

File: sandbox/src/mocks/getReferralRequest/responses/Unbooked.json [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        "coding": [
+          {
+            "system": "http://fhir.nhs.uk/ValueSet/ers-attachmenttype-1",
+            "code": "REFERRER"
+          }
+        ]
+      },
+      "status": "current",
+      "indexed": "ignoreDate",
+      "description": "",
```
Comment (nhsd-mark-warren) [inline sandbox/src/mocks/getReferralRequest/responses/Unbooked.json:44]: Do we need a more realistic values for this description (again, I just ripped this from the XAPI IT)

File: scripts/populate_placeholders.py [hunk 7]
```diff
@@ -19,6 +19,7 @@ def main():
       "[[HYPERLINK_A012]]": "[A012 - Maintain referral letter](https://developer.nhs.uk/apis/e-Referrals/explore_endpoint_a012.html)",
       "[[HYPERLINK_A015]]": "[A015 - Retrieve appointment slots](#api-Default-a015-retrieve-appointment-slots)",
       "[[HYPERLINK_A016]]": "[A016 - Book or defer appointment](https://developer.nhs.uk/apis/e-Referrals/explore_endpoint_a016.html)",
+      "[[HYPERLINK_A019]]": "[A019 - Generate patient letter](https://developer.nhs.uk/apis/e-Referrals/explore_endpoint_a019.html)",
```
Comment (nhsd-hubert-raich) [inline scripts/populate_placeholders.py:22]: Also update the link for A005 to point to : '#api-Default-a005-retrieve-referral-request'. The url path is build from '#api-TAG-OPERATION-ID'

File: sandbox/src/mocks/getReferralRequest/responses/Unbooked.json (hunk 1 above)
Comment (nhsd-mark-warren) [reply sandbox/src/mocks/getReferralRequest/responses/Unbooked.json:11]: IIRC Hubert said there was no particular order.

File: sandbox/src/mocks/getReferralRequest/responses/Unbooked.json (hunk 3 above)
Comment (nhsd-mark-warren) [reply sandbox/src/mocks/getReferralRequest/responses/Unbooked.json:27]: Discussed with Hubert. There isn't.

File: sandbox/src/mocks/getReferralRequest/responses/Unbooked.json [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+      "content": [
+        {
+          "attachment": {
+            "id": "70002",
+            "extension": [
+              {
+                "url": "http://fhir.nhs.uk/StructureDefinition/extension-ers-attachedby-1",
+                "valueReference": {
+                  "reference": "#Practitioner-021600556514"
```
[the remark this answers is not in the document]
Comment (nhsd-mark-warren) [reply sandbox/src/mocks/getReferralRequest/responses/Unbooked.json:53]: Discussed with Hubert. We don't.

File: specification/e-referrals-service-api.yaml (hunk 5 above)
Comment (nhsd-mark-warren) [reply specification/e-referrals-service-api.yaml:318]: Removed from the yaml file. I see https://nhsd-jira.digital.nhs.uk/browse/ERSSUP-54848 has been updated too, so we're aligned.

File: scripts/populate_placeholders.py (hunk 7 above)
Comment (nhsd-mark-warren) [reply scripts/populate_placeholders.py:22]: Udpated

**Comment by nhsd-hubert-raich** [comment]
Closing this PR as it will be replaced by of of mine (some work resused from Mark's branch)

**Pull request closed without merging**